File: mario.c
#include <stdio.h>
#include <cs50.h>

int repeat(char c, int n);

int main(void)
{
    int pyramidHeight = 0;
    do
    {
        pyramidHeight = get_int("Half-pyramid height: ");
    }
    while (0 > pyramidHeight || pyramidHeight > 23);
    int line = 1;
    for (int i = pyramidHeight; i > 0; i--)
    {
        repeat(' ', pyramidHeight - line);
        repeat('#', line);
        printf("  ");
        repeat('#', line);
        printf("\n");
        line++;
    }
}

int repeat(char c, int n)
{
    for (int i = 0; i < n; i++)
    {
        printf("%c", c);
    }
    return 1;
}

File: mario.py
from cs50 import get_int

# loop until the number of the pyramidHeight is in the range 0-23
while True:
    pyramidHeight = get_int("Pyramid Height: ")
    if 0 <= pyramidHeight and pyramidHeight <= 23:
        break
base = pyramidHeight + 1
line = 2
for i in range(pyramidHeight):
    text = ' ' * (base - line) + '#' * line
    print(text)
    line += 1